examples/plugins/altemplate/src/log.rs

```diff
@@ -0,0 +1,84 @@
+/* Copyright (C) 2018 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+// same file as rust/src/applayertemplate/logger.rs except
+// different paths for use statements
+// open_object using altemplate instead of just template
+// Jsonbuilder using C API due to  opaque implementation
+
+use super::template::TemplateTransaction;
+use std::ffi::{c_char, CString};
+use suricata::cast_pointer;
+use suricata::jsonbuilder::JsonError;
+
+use std;
+
+// Jsonbuilder opaque with implementation using C API to feel like usual
+#[repr(C)]
+pub struct JsonBuilder {
```

Comment:
Have you tried exporting the C JsonBuilder API with bindgen?  I did try a while back and it didn't work. But didn't try after getting the bindgen work committed.

To recommend a template/example plugin for people to use, I don't think it can be providing any bindings to core Suricata functions, those should all come from Suricata. As it puts the burden on the plugin developer to make sure those bindings are up to date, and not use.

Reply:
> To recommend a template/example plugin for people to use, I don't think it can be providing any bindings to core Suricata functions, those should all come from Suricata. As it puts the burden on the plugin developer to make sure those bindings are up to date, and not use.

I see the point

> Have you tried exporting the C JsonBuilder API with bindgen?

I do not understand.
The "C" JsonBuilder API is in rust like `pub unsafe extern "C" fn jb_set_string`

But what we want here is Suricata to provide a wrapper that looks like rust even it goes through C API under the hood...

Reply:
So, I guess I need to start the new `suricata-core` crate (or a better name) for 0 dependency and use by plugins, and include the JsonBuilder wrapper that is in this PR in this crate

Reply:
Bindgen `0.71.1` has no issues re-exporting the generated rust-bindings.h back to Rust.

Reply:
> At a minimum we need to get bindings for the C JsonBuilder API back into Rust right?

I do not see why... Can you expand ?

> On naming, core sounds like it would be suricata core features. What you describe is a crate to be used by things linking to Suricata? Maybe suricata_extern or something?

It is also used by Suricata itself, see my next PR

Reply:
> I do not see why... Can you expand ?
> 
> >

This extern C block: https://github.com/OISF/suricata/pull/12886/files#diff-b7126dc0c8e2a6d3d13093dfef151fda3de85ebe9820a2381360f267d822e92dR32

But will comment more over there.